uniref-rest/src/test/java/org/uniprot/api/uniref/controller/UniRefGetIdControllerIT.java

```diff
@@ -91,110 +90,199 @@ protected String getIdRequestPath() {
+                .andExpect(jsonPath("$.memberCount", is(15)));
     }
 
-    private RepresentativeMember createReprestativeMember() {
-        String seq = "MVSWGRFICLVVVTMATLSLARPSFSLVEDDFSAGSADFAFWERDGDSDGFDSHSDJHETRHJREH";
-        Sequence sequence = new SequenceBuilder(seq).build();
-        String memberId = "P12345_HUMAN";
-        int length = 312;
-        String pName = "some protein name";
-        String upi = "UPI0000083A08";
-
-        UniRefMemberIdType type = UniRefMemberIdType.UNIPROTKB;
-
-        return new RepresentativeMemberBuilder()
-                .memberIdType(type)
-                .memberId(memberId)
-                .organismName("Homo sapiens")
-                .organismTaxId(9606)
-                .sequenceLength(length)
-                .proteinName(pName)
-                .uniparcId(new UniParcIdBuilder(upi).build())
-                .accessionsAdd(new UniProtKBAccessionBuilder("P12345").build())
-                .uniref100Id(new UniRefEntryIdBuilder("UniRef100_P03923").build())
-                .uniref90Id(new UniRefEntryIdBuilder("UniRef90_P03943").build())
-                .uniref50Id(new UniRefEntryIdBuilder("UniRef50_P03973").build())
-                .isSeed(true)
-                .sequence(sequence)
-                .build();
+    @Test
+    void getLastPageMembers() throws Exception {
+        // given
+        UniRefEntry entry = UniRefControllerITUtils.createEntry(1, 15, UniRefType.UniRef50);
+        saveEntry(entry);
+
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                get(getIdRequestPath() + ID)
+                                        .param("size", "10")
+                                        .param("cursor", "3sbq7rwffis")
+                                        .header(ACCEPT, APPLICATION_JSON_VALUE));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(header().string("X-TotalRecords", "15"))
+                .andExpect(header().string(HttpHeaders.LINK, nullValue()))
+                .andExpect(jsonPath("$.id").doesNotExist())
+                .andExpect(jsonPath("$.representativeMember").doesNotExist())
+                .andExpect(jsonPath("$.memberCount").doesNotExist())
+                .andExpect(jsonPath("$.members.size()", is(5)))
+                .andExpect(
+                        jsonPath(
+                                "$.members[*].memberId",
+                                contains(
+                                        "P32110_HUMAN",
+                                        "P32111_HUMAN",
+                                        "P32112_HUMAN",
+                                        "P32113_HUMAN",
+                                        "P32114_HUMAN")));
     }
```

Comment:
Like i have commented in class UniRefEntryStoreRepositoryTest.java , we should add at least one negative test case around size.

Reply:
Can you please elaborate an use case?

Reply:
Same as comment in UniRefEntryStoreRepositoryTest
"lets say a cluster has only 10 members but the client is passing size as 100.
A cluster has only a rep member and user is passing size as more than 1."